src/irr/asset/CGLSLBRDFBuiltinIncludeLoader.h

```diff
@@ -641,13 +774,170 @@ vec3 irr_glsl_ggx_height_correlated_cos_eval(in irr_glsl_BSDFIsotropicParams par
 #include <irr/builtin/glsl/bsdf/brdf/specular/geom/smith.glsl>
 #include <irr/builtin/glsl/bsdf/brdf/specular/fresnel/fresnel.glsl>
 
+//i wonder where i got irr_glsl_ggx_smith_height_correlated() from because it looks very different from 1/(1+L_v+L_l) form
+float irr_glsl_beckmann_smith_height_correlated(in float NdotV2, in float NdotL2, in float a2)
+{
+    float c2 = irr_glsl_smith_beckmann_C2(NdotV2, a2);
+    float L_v = irr_glsl_smith_beckmann_Lambda(c2);
+    c2 = irr_glsl_smith_beckmann_C2(NdotL2, a2);
+    float L_l = irr_glsl_smith_beckmann_Lambda(c2);
+    return 1.0 / (1.0 + L_v + L_l);
+}
+
+//idk what erf even is but PBRT's implementation of beckmann sampling uses it..
```

Comment:
erf is the indefinite integral of `e^(-x^2)`
![image](https://user-images.githubusercontent.com/6894321/83310109-47226d80-a20b-11ea-9dd0-1bde764b7b89.png)

Reply:
these functions better go into some C++ STD-like header

Reply:
done?

Reply:
it already is in std https://en.cppreference.com/w/cpp/numeric/math/erf